datafusion/physical-expr/src/expressions/negative.rs

```diff
@@ -164,9 +164,9 @@ pub fn negative(
         && !is_interval(&data_type)
         && !is_timestamp(&data_type)
     {
-        internal_err!(
-            "Can't create negative physical expr for (- '{arg:?}'), the type of child expr is {data_type}, not signed numeric"
-        )
+        Err(DataFusionError::Plan(
+            "Negation only supports numeric, interval and timestamp types".to_string(),
+        ))
```

Comment:
You can also use `plan_err!` here for less verbose code

```suggestion
        plan_err!("Negation only supports numeric, interval and timestamp types")
```